### DISCUSSION TITLE
Windows BAT problem with imagemagick / imageJ

### DISCUSSION THREAD

**Discussion by Jossie-X** [ImageMagick/ImageMagick#3927] [Help]
Good evening,

sorry, but I have narrowed down the problem and edited the original post:

I want to extract the first image with this command, but it gets stuck:

`magick 141_(21)_64bit_HDRi(1).tif[0] 141_(21)_64bit_HDRi(1)_0.tif`

If I extract the second image [1]
`magick 141_(21)_64bit_HDRi(1).tif 141_(21)_64bit_HDRi(1)_0.tif`
 
or the third [2]
`magick 141_(21)_64bit_HDRi(1).tif[2] 141_(21)_64bit_HDRi(1)_2.tif`

everything works fine.

I cannot figure out, what is wrong with the first command.

Any hint would be highly appreciated!

Hermann-Josef

**Comment by snibgo** [comment]
Your edit removed important information: What version of IM, on what platform?

Please link to your file 141_(21)_64bit_HDRi(1).tif. I suspect the first image is strange. Perhaps it is too large for your memory, so IM uses the disk, which is much slower, giving the impression it is "stuck".

**Reply by Jossie-X** [reply]
Hello,

thanks a lot for the fast reply.

In the meantime I made some progress.

Sorry, this was omitted by the edit: Imagemagick version is 7.1.0-Q16.

I do not think it is a memory problem, since it work fine with the very same image before I had to set up my PC from scratch, and installed the latest imagemagick version. The latter is the only change!

I checked with EXIFtool, that there are indeed 3 images there.

If I extract the first one, it works fine. If I extract the second one, the 3. image is extraced. And if I try to extract the third image, an error is occuring:

`D:\Hermann-Josef\Pictures\Tests>magick Dia_1410_Kodachrome.tif[0] Dia_1410_Kodachrome_0.tif
magick: Unknown field with tag 35243 (0x89ab) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/960.

D:\Hermann-Josef\Pictures\Tests>magick Dia_1410_Kodachrome.tif[1] Dia_1410_Kodachrome_1.tif
magick: Unknown field with tag 35243 (0x89ab) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/960.

D:\Hermann-Josef\Pictures\Tests>magick Dia_1410_Kodachrome.tif[2] Dia_1410_Kodachrome_2.tif
magick: Unknown field with tag 35243 (0x89ab) encountered. `TIFFReadDirectory' @ warning/tiff.c/TIFFWarnings/960.
magick: no images for write '-write' 'Dia_1410_Kodachrome_2.tif' at CLI arg 2 @ error/operation.c/CLINoImageOperator/4894.`

So I am really puzzled, even more due to the fact that my imageJ-macro, which calls this BAT file, worked fine (no changes made to the code) before I had to make a complete re-installation of my PC.

Hermann-Josef

Here you can find the image I am now testing this with: https://adobe.ly/2uYFMge  (It is large, sorry. But I hesitate to fiddle with the image in order not to create another problem.).

This is a slide scanned with Silverfast. The first image in the TIF is the full resolution scan, the second the low-resolution preview image, the third image is the (full resolution) IR-channel.

**Comment by snibgo** [comment]
> I checked with EXIFtool, that there are indeed 3 images there.

What exiftool command shows 3 images in that file?

IM can see only two images:
```
magick Dia_1410_Kodachrome.tif info:

magick: Unknown field with tag 35243 (0x89ab) encountered. `TIFFReadDirectory' @  warning/tiff.c/TIFFWarnings/1017.
Dia_1410_Kodachrome.tif[0] TIFF 7100x4751 7100x4751+0+0 16-bit TrueColor sRGB 257.684MiB 1.781u 0:01.793
Dia_1410_Kodachrome.tif[1] TIFF 7100x4751 7100x4751+0+0 16-bit Grayscale Gray 0.406u 0:00.438
```
Note that they are the same size, but image [1] is grayscale. So extracting images [0] and [1] should work, but [2] will fail.

The warning is not important.

With IM v7.0.8-64, I have no problem reading those two images:
```
magick Dia_1410_Kodachrome.tif[0] x0.tiff

magick Dia_1410_Kodachrome.tif[1] x1.tiff
```

**Reply by Jossie-X** [reply]
I see the tags with EXIFtool: IFD0 (full resolution), IFD1 (low resolution) and IFD2 (transparency mask.

As I said, I used this image previously and evrything worked fine. Only change is the new version of IM. So my strong suspicion of course is a change in the behaviour of IM.

Could it be that IM does change the input image? I mean, remove the first image, after it is extracted? Then if I extract the second image in the second command, I get the IR image and so forth? well, just an idea.

The second image is of smaller size. Can this be a problem?

The grey scale image is the 3. image in the original TIF - the IR channel. This is how SilverFast is set up.

Hermann-Josef

**Comment by snibgo** [comment]
According to exiftool, the file contains only two images:
```
exiftool -a -u -g1 Dia_1410_Kodachrome.tif
```
The output, removing irrelevant profile data, is:
```
---- ExifTool ----
ExifTool Version Number         : 12.13
---- System ----
File Name                       : Dia_1410_Kodachrome.tif
Directory                       : .
File Size                       : 258 MiB
File Modification Date/Time     : 2021:07:13 18:46:44+01:00
File Access Date/Time           : 2021:07:13 18:46:19+01:00
File Creation Date/Time         : 2021:07:13 18:46:19+01:00
File Permissions                : rw-rw-rw-
---- File ----
File Type                       : TIFF
File Type Extension             : tif
MIME Type                       : image/tiff
Exif Byte Order                 : Big-endian (Motorola, MM)
Current IPTC Digest             : 6f45959c761f9be9eb4fecdcb0313040
---- IFD0 ----
Subfile Type                    : Full-resolution image
Image Width                     : 7100
Image Height                    : 4751
Bits Per Sample                 : 16 16 16
Compression                     : Uncompressed
Photometric Interpretation      : RGB
Make                            : Reflecta
Camera Model Name               : MS Scanner
Strip Offsets                   : (Binary data 46753 bytes, use -b option to extract)
Orientation                     : Horizontal (normal)
Samples Per Pixel               : 3
Rows Per Strip                  : 1
Strip Byte Counts               : (Binary data 28505 bytes, use -b option to extract)
X Resolution                    : 5000
Y Resolution                    : 5000
Planar Configuration            : Chunky
Resolution Unit                 : inches
Software                        : SilverFast 8.8.0 r14 (Aug 31 2018)  88cdfc4 31.08.
Exif 0x89ab                     : 12 20 231 194 0 0 0 0
---- XMP-x ----
XMP Toolkit                     : XMP Core 4.4.0-Exiv2
---- XMP-photoshop ----
Source                          : Kodachrome
---- IPTC ----
Source                          : Kodachrome
Date Created                    : 2018:09:28
---- ExifIFD ----
Maker Note Unknown Binary       : (Binary data 130 bytes, use -b option to extract)
Exif Image Width                : 7100
Exif Image Height               : 4751
---- ICC-header ----
Profile CMM Type                : Unknown (argl)
Profile Version                 : 2.2.0
Profile Class                   : Input Device Profile
Color Space Data                : RGB
Profile Connection Space        : Lab
Profile Date Time               : 2020:11:25 17:01:18
Profile File Signature          : acsp
Primary Platform                : Microsoft Corporation
CMM Flags                       : Not Embedded, Independent
Device Manufacturer             : 
Device Model                    : 
Device Attributes               : Transparency, Glossy, Positive, Color
Rendering Intent                : ICC-Absolute Colorimetric
Connection Space Illuminant     : 0.9642 1 0.82491
Profile Creator                 : Unknown (argl)
Profile ID                      : 0
---- ICC_Profile ----
Profile Description             : Kodachrome_AV150_bin_g22 a h l u 2020-11-25 17:01_p3.00
Profile Copyright               : Hermann-Josef Roeser
Device Mfg Desc                 : reflecta
Device Model Desc               : DigitDia6000
Media White Point               : 0.6577 0.71706 0.6324
Media Black Point               : 0.00298 0.00325 0.00386
A To B0                         : (Binary data 240250 bytes, use -b option to extract)
Char Target                     : (Binary data 24527 bytes, use -b option to extract)
{...profile data removed...}
ICC Profile Arts                : 0.8951 0.2664 -0.16141 -0.7502 1.7135 0.0367 0.03889 -0.0685 1.0296
---- IFD1 ----
Subfile Type                    : Transparency mask
Image Width                     : 7100
Image Height                    : 4751
Bits Per Sample                 : 16
Compression                     : Uncompressed
Photometric Interpretation      : BlackIsZero
Strip Offsets                   : (Binary data 42031 bytes, use -b option to extract)
Samples Per Pixel               : 1
Rows Per Strip                  : 1
Strip Byte Counts               : (Binary data 28505 bytes, use -b option to extract)
X Resolution                    : 5000
Y Resolution                    : 5000
Planar Configuration            : Chunky
Resolution Unit                 : inches
---- Composite ----
Image Size                      : 7100x4751
Megapixels                      : 33.7
```
IM won't read preview images. This file seems to contain no preview images.

According to both IM and exiftool, the two images have the same number of pixels: 7100x4751.

Provided your output filename is different to the input filename, IM won't alter input files.

**Comment by Jossie-X** [comment]
You are right, it is only 2 images. Mea culpa, mea maxima culpa!

Same name, two different directories, one had the preview embedded, the other not. So I got totaly confused.

Okay, thanks a lot for looking into this.

I still have a problem with another image, but this is not mine, so I have first to find out, if it is compatible with my macro.

I hope I can get this sorted out by myself then.

Many thanks again and best wishes

Hermann-Josef

**Comment by snibgo** [comment]
No problem. Do come back if you need help.
